cmake/onnxruntime_mlas.cmake

```diff
@@ -631,6 +631,10 @@ if (WIN32)
   endif()
 endif()
 
+if (PLATFORM_NAME STREQUAL "macabi")
+  target_compile_options(onnxruntime_mlas PRIVATE ${CMAKE_C_FLAGS}) # Needed for maccatalyst C compilation
```

Comment:
Not sure what else is in CMAKE_C_FLAGS. 

Can we limit it to specifying `-Wno-overriding-t-option` directly or are there other flags required?

Reply:
probably not, this set CMAKE_C_FLAGS contains three args:
e.g.
"--target=x86_64-apple-ios14.0-macabi -ffunction-sections -fdata-sections"

expanded the comment here for notes.